### PR TITLE
8188784: javax/management/notification/BroadcasterSupportDeadlockTest.java - TEST FAILED: deadlock

### PR DESCRIPTION
**Pull request by kevinjwalls**
This test incorrectly fails, although rarely, thinking its "thread 2" has deadlocked.
A change of sleep will likely fix this, but there are other issues, so cleaning up the test a little.

Remove the probe for the ManagementFactory class, to check we are on jdk5 or later. 8-)

When sleeping, sleep 100, not 1ms, we don't need to spin fast and actually race with the other thread.

We have a 1000 iteration loop, but don't seem to use it.  We only check once then either return (pass), fail, or break (which is also fail).  Use the loop to check for the status change, which is likely what was intended.

Show the stackframes on all failures.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8188784](https://bugs.openjdk.org/browse/JDK-8188784): javax/management/notification/BroadcasterSupportDeadlockTest.java - TEST FAILED: deadlock (**Bug** - P4)

### Reviewers
 * [Chris Plummer](https://openjdk.org/census#cjplummer) (@plummercj - **Reviewer**) ⚠️ Review applies to [5237f1d8](https://git.openjdk.org/jdk/pull/18687/files/5237f1d8ad936e4c98c1c028eaf64eb0f9eee279)
 * [Leonid Mesnik](https://openjdk.org/census#lmesnik) (@lmesnik - **Reviewer**) ⚠️ Review applies to [5237f1d8](https://git.openjdk.org/jdk/pull/18687/files/5237f1d8ad936e4c98c1c028eaf64eb0f9eee279)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/18687/head:pull/18687` \
`$ git checkout pull/18687`

Update a local copy of the PR: \
`$ git checkout pull/18687` \
`$ git pull https://git.openjdk.org/jdk.git pull/18687/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 18687`

View PR using the GUI difftool: \
`$ git pr show -t 18687`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/18687.diff">https://git.openjdk.org/jdk/pull/18687.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/18687#issuecomment-2044813296)

### CODE REVIEW

File: test/jdk/javax/management/notification/BroadcasterSupportDeadlockTest.java [hunk 1]
```diff
@@ -105,31 +101,25 @@ public void run() {
...
-        System.out.println("TEST FAILED BUT DID NOT NOTICE DEADLOCK");
-        Thread.sleep(10000);
+        if (t2.getState() == Thread.State.BLOCKED) {
+            System.out.println("TEST FAILED: deadlock");
+        } else {
+            System.out.println("TEST FAILED BUT DID NOT NOTICE DEADLOCK (state = " + t2.getState() + ")");
+        }
+        java.util.Map<Thread,StackTraceElement[]> traces = Thread.getAllStackTraces();
+        showStackTrace("Thread 1", traces.get(t1));
+        showStackTrace("Thread 2", traces.get(t2));
```
Comment (lmesnik) [inline test/jdk/javax/management/notification/BroadcasterSupportDeadlockTest.java:122]: Could you please replace System.exit() with throwing Exception. Other looks good.

**Review by lmesnik — APPROVED** [review]

File: test/jdk/javax/management/notification/BroadcasterSupportDeadlockTest.java (hunk 1 above)
Comment (kevinjwalls) [reply test/jdk/javax/management/notification/BroadcasterSupportDeadlockTest.java:121]: OK sure.
FYI I count 137 System.exit calls in test/jdk/javax/management
That's after I take out the existing 3 that are in this test. 8-)

**Comment by kevinjwalls** [comment]
Thanks for the reviews!

**Pull request closed without merging**